go/app/main.go

```
@@ -28,14 +43,113 @@ func root(c echo.Context) error {
+
+	// Get the final hash value
+	hashValue := hash.Sum(nil)
+	// Convert the byte slice to a hex-encoded string
+	hashString := hex.EncodeToString(hashValue)
+	image = hashString + ".jpg"
+
+	var res Response
+	var itemslice []Item
+	// Create a new item with the next ID
+	newItem := Item{
+		Name:     name,
+		Category: category,
+		Image:    image,
+	}
+
+	if _, err := os.Stat("items.json"); err == nil {
+		file1, err := os.Open("items.json") //すでにあるファイルを開く
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		defer file1.Close()
+		jsonData, err := ioutil.ReadAll(file1)
+		if err != nil {
+			fmt.Println("JSONデータを読み込めません", err)
+		}
 
-	message := fmt.Sprintf("item received: %s", name)
-	res := Response{Message: message}
+		json.Unmarshal(jsonData, &itemslice)
+
+		//Print message
+		message := fmt.Sprintf("item received: %s in %s category", newItem.Name, newItem.Category)
+		res = Response{Message: message}
+
+		// Append the new item to the slice
+		itemslice = append(itemslice, newItem)
+
+	} else {
+		itemslice = append(itemslice, newItem)
+
+	}
+
+	file2, err := os.Create("items.json") // fileはos.File型
+	if err != nil {
+		log.Fatal(err)
+	}
+	json.NewEncoder(file2).Encode(itemslice)
 
 	return c.JSON(http.StatusOK, res)
 }
 
+func getItems(c echo.Context) error {
+	file1, err := os.Open("items.json") //すでにあるファイルを開く
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file1.Close()
+	jsonData, err := ioutil.ReadAll(file1)
+	if err != nil {
+		fmt.Println("JSONデータを読み込めません", err)
+	}
+	var itemslice []Item
+	json.Unmarshal(jsonData, &itemslice)
+	fmt.Println(itemslice)
+
+	return c.JSON(http.StatusOK, itemslice)
+}
+
+func getItem(c echo.Context) error {
+	itemIDStr := c.Param("item_id")
+
+	itemID, err := strconv.Atoi(itemIDStr)
+	if err != nil {
+		// エラーが発生した場合の処理
+		fmt.Println("item_idを整数に変換できません:", err)
+	}
+
+	fmt.Println(itemID)
+
+	file1, err := os.Open("items.json") //すでにあるファイルを開く
+	if err != nil {
+		log.Fatal(err)
```

We need to be careful with `log.Fatal`. If it is called, our server will stop and nobody will be able to access it anymore 😅 . So it is better to return the error so the server continues to run.